### DISCUSSION TITLE
mqtt messagebus for exgex v2.3.0 issue

### DISCUSSION THREAD

**Discussion by edgego** [edgexfoundry/community#168] [General]
Hi team , I used mqtt messagebus , but from rule engine ekuiper, can not subscribe to messagebus, please help!!!
time="2023-08-18 02:26:36" level=error msg="Failed to subscribe to edgex messagebus with topic edgex/events/device/Test-Device-Modbus-Profile/mod1# has error : An error occured while performing a 'Subscribe' operation: connection lost before Subscribe completed." file="edgex/edgex.go:143"
time="2023-08-18 02:26:36" level=error msg="Failed to subscribe to edgex messagebus topic An error occured while performing a 'Subscribe' operation: connection lost before Subscribe completed.\n" file="edgex/edgex_source.go:94" source_pool=edgex.mod1
here is the data source config:
mod1:
        messageType: request
        optional:
            ClientId: kuiper-rules-engine
            KeepAlive: 500
            Qos: 0
            Retained: false
            SkipCertVerify: true
        port: 1883
        protocol: tcp
        server: edgex-mqtt-broker
        topic: edgex/events/device/Test-Device-Modbus-Profile/mod1#
        type: mqtt

**Comment by ngjaying** [comment]
Please check https://github.com/lf-edge/ekuiper/issues/2197#issuecomment-1683575751

**Comment by lenny-goodell** [comment]
closing as answered per comments on https://github.com/lf-edge/ekuiper/issues/2197#issuecomment-1683575751
